src/core/orbital-factory.ts

```diff
@@ -1,39 +1,23 @@
-import { isFunction } from 'util';
-
-import { CLIMetadata } from './decorators/cli/cli-metadata';
 import { CommandExecutor } from './command-executor';
 import { CommandMapper } from './command-mapper';
-import { Constructor } from './util/constructor';
-import { Executable } from './interfaces/executable';
-import { arrayIsPopulated } from './util/array';
-import { getClassMetadata } from './reflection/class';
 import { CommandParser } from './command-parser';
+import { CLIMetadata } from './decorators/cli/cli-metadata';
+import { getClassMetadata } from './reflection/class';
+import { arrayIsPopulated } from './util/array';
+import { Constructor } from './util/constructor';
 
-// TODO: Why Append static ?!!
+// WBH
+// TODO: Review this. Probably not a good idea to chain functions this much,
```

Comment:
I actually prefer having two differents calls. One is for creating the CLI which IMO should still return an executable and not the factory. And then you can execute string arguments with the created CLI.

Reply:
actually that's an intresting idea. Let's work on designing an api for that